cloud/openstack/os_quota.py

```diff
@@ -0,0 +1,339 @@
+        injected_files=dict(required=False, type='int', default=None),
+        injected_path_size=dict(required=False, type='int', default=None),
+        instances=dict(required=False, type='int', default=None),
+        key_pairs=dict(required=False, type='int', default=None),
+        network=dict(required=False, type='int', default=None),
+        per_volume_gigabytes=dict(required=False, type='int', default=None),
+        port=dict(required=False, type='int', default=None),
+        project=dict(required=False, type='int', default=None),
+        properties=dict(required=False, type='int', default=None),
+        ram=dict(required=False, type='int', default=None),
+        rbac_policy=dict(required=False, type='int', default=None),
+        router=dict(required=False, type='int', default=None),
+        secgroup_rules=dict(required=False, type='int', default=None),
+        secgroups=dict(required=False, type='int', default=None),
+        server_group_members=dict(required=False, type='int', default=None),
+        server_groups=dict(required=False, type='int', default=None),
+        snapshots=dict(required=False, type='int', default=None),
+        snapshots_types=dict(required=False, type='dict', default={}),
+        subnet=dict(required=False, type='int', default=None),
+        subnetpool=dict(required=False, type='int', default=None),
+        volumes=dict(required=False, type='int', default=None),
+        volumes_types=dict(required=False, type='dict', default={})
+    )
+
+
+    module = AnsibleModule(argument_spec)
+
+    if not HAS_SHADE:
+        module.fail_json(msg='shade is required for this module')
+
+    try:
+        cloud_params = dict(module.params)
+        cloud = shade.operator_cloud(**cloud_params)
+
+        #In order to handle the different volume types we updated module params after.
+        dynamic_types = [
+            'gigabytes_types',
+            'snapshots_types',
+            'volumes_types',
+        ]
+
+        #import epdb
+        #epdb.serve()
+        for dynamic_type in dynamic_types:
+            for k, v in module.params[dynamic_type].iteritems():
+                module.params[k] = int(v)
+
+        #Get current quota values
+        project_quota_output = _get_quotas(cloud, cloud_params['name'])
+
+        quota_change_request = {}
+
+        changes_required = False
+        for quota_type in project_quota_output.keys():
+            for quota_option in project_quota_output[quota_type].keys():
+                if quota_option in module.params and module.params[quota_option] is not None:
+                    if project_quota_output[quota_type][quota_option] != module.params[quota_option]:
+                        changes_required = True
+
+                        if quota_type not in quota_change_request:
+                            quota_change_request[quota_type] = {}
+
+                        quota_change_request[quota_type][quota_option] = module.params[quota_option]
```

Comment:
This should be moved to a function like the other os_ modules use to detect if change is needed.